### DISCUSSION TITLE
Support for null-safe equals operator <=>?

### DISCUSSION THREAD

**Discussion by underminedsk** [JSQLParser/JSqlParser#1487] [Q&A] [answered]
Hi, I have been testing out your parser for a project and overall it works great but I found that this simple statement is not parsable: 
```
CCJSqlParserUtil.parse("select * from foo where bar <=> 'baz'");
```
throws the exception
```
net.sf.jsqlparser.JSQLParserException: Encountered unexpected token: "<=" <OP_MINORTHANEQUALS>
    at line 1, column 29.
```
The null-safe equals operator is supported in many commonly used SQL systems, including MariaDB, Snowflake, and Spark SQL.  Did I not configure the parser correctly, or there a way to adjust the grammar dialect to handle this operator?   Please advise.

Thanks!
Zach

**Comment by manticore-projects** [comment, marked as answer]
Greetings!

> The null-safe equals operator is supported in many commonly used SQL systems, including MariaDB, Snowflake, and Spark SQL. 

JSQLParser is largely RDBMS agnostic with a certain preference to SQL Standard and the Oracle, MS SQL Server and PostgreSQL Dialects.

> Did I not configure the parser correctly,

No, except `SQUARED_BRACKET_QUOTATION`, there is no configuration needed. (And this does not apply here.)
Especially, there is no configuration of "Dialects".

> or there a way to adjust the grammar dialect to handle this operator?

The only ways to make it work were:
a) Submit a PR implementing this operator (Grammar, Java Code, Tests) 
b) OR sponsor an implementation

Best regards
Andreas
